tutorials/src/bin/basic-tutorial-5.rs

```diff
@@ -0,0 +1,377 @@
+                }
+
+                unsafe {
+                    let xid = gdk_x11_window_get_xid(gdk_window.to_glib_none().0);
+                    video_overlay.set_window_handle(xid as usize);
+                }
+            } else {
+                println!("Add support for display type '{}'", display_type_name);
+                process::exit(-1);
+            }
+        } else if cfg!(feature = "tutorial5-quartz") {
+            if display_type_name == "GdkQuartzDisplay" {
+                extern "C" {
+                    pub fn gdk_quartz_window_get_nsview(
+                        window: *mut glib::object::GObject,
+                    ) -> *mut c_void;
+                }
+
+                unsafe {
+                    let window = gdk_quartz_window_get_nsview(gdk_window.to_glib_none().0);
+                    video_overlay.set_window_handle(window as usize);
+                }
+            } else {
+                println!(
+                    "Unsupported display type '{}', compile with `--feature `",
+                    display_type_name
+                );
+                process::exit(-1);
+            }
+        }
+    });
+
+    let streams_list = gtk::TextView::new();
+    streams_list.set_editable(false);
+    let pipeline = playbin.clone();
+    let textbuf = SendCell::new(
+        streams_list
+            .get_buffer()
+            .expect("Couldn't get buffer from text_view"),
+    );
+    playbin.get_bus().unwrap().connect_message(move |_, msg| {
+        //let streams_view = &streams_view;
+        match msg.view() {
+            gst::MessageView::Application(_) => if msg.get_structure().get_name() == "tags-changed"
+            {
+                analyze_streams(&pipeline, &textbuf);
+            },
+            _ => (),
+        }
+    });
+
+    let vbox = Box::new(Orientation::Horizontal, 0);
+    vbox.pack_start(&video_window, true, true, 0);
+    vbox.pack_start(&streams_list, false, false, 2);
+
+    let main_box = Box::new(Orientation::Vertical, 0);
+    main_box.pack_start(&vbox, true, true, 0);
+    main_box.pack_start(&controls, false, false, 0);
+    main_window.add(&main_box);
+    main_window.set_default_size(640, 480);
+
+    main_window.show_all();
+}
+
+#[cfg(feature = "tutorial5")]
+// We are possibly in a GStreamer working thread, so we notify the main
+// thread of this event through a message in the bus
+fn post_app_message(playbin: &gst::Element) {
+    let mbuilder = gst::Message::new_application(gst::Structure::new_empty("tags-changed"));
+    playbin.post_message(&mbuilder.build());
+}
+
+#[cfg(feature = "tutorial5")]
+fn main() {
+    // Make sure the right features were activated
+    if !cfg!(feature = "tutorial5-x11") && !cfg!(feature = "tutorial5-quartz") {
+        eprintln!("No Gdk backend selected, compile with --features tutorial5[-x11][-quartz].");
+
+        return;
+    }
+
+    // Initialize GTK
+    if gtk::init().is_err() {
+        eprintln!("Failed to initialize GTK.");
```

Comment:
`if let Err(err) = gtk::init()` and then also print the error. Same for GStreamer. `is_err`, `is_some` and all these are usually not nice

Reply:
Done.